src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (this.hidden) {
+      this.hidden = false;
+    } else {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const victimIndex = Animal.alive.findIndex(animal => animal === victim);
+
+    if (victim.hasOwnProperty('hidden') && victim.hidden === false) {
```

Comment:
You are checking if the victim is a Herbivore by checking the 'hidden' property. However, it would be better to check the instance of the object to ensure the victim is a Herbivore. You can use 'instanceof' operator to check if the victim is an instance of Herbivore class.